sentry-android-core/src/main/java/io/sentry/android/core/AndroidContinuousProfiler.java

```diff
@@ -58,6 +58,7 @@ public class AndroidContinuousProfiler
   private @NotNull SentryDate startProfileChunkTimestamp = new SentryNanotimeDate();
   private boolean shouldSample = true;
   private boolean isSampled = false;
+  private int rootSpanCounter = 0;
```

Comment:
```suggestion
  private volatile int rootSpanCounter = 0;
```

Reply:
don't think we need it, as we access it only in synchronized blocks